javatests/arcs/core/storage/database/BUILD

```diff
@@ -0,0 +1,18 @@
+load("//tools/build_defs/kotlin:rules.bzl", "kt_jvm_library")
+
+licenses(["notice"])
+
+package(default_visibility = ["//visibility:public"])
+
+kt_jvm_library(
+    name = "mock-database",
```

Comment:
Test helpers generally go in a "testing" package under "java", not "javatests".

//java/arcs/core/storage/database/testing

(Don't forget to set testonly=True)

Reply:
Yeah, but I didn't want to put it there because it requires jvm-only stuff (mockito)..

Reply:
I'll move it to `//java/arcs/jvm/..../testing` in my next PR (actual driver impl, where the mock will be more important to have anyway)